librz/analysis/var.c

```diff
@@ -117,21 +133,16 @@ RZ_API RzAnalysisVar *rz_analysis_function_set_var(RzAnalysisFunction *fcn, int
 	}
 	var->name = strdup(name);
 	var->regname = reg ? strdup(reg->name) : NULL; // TODO: no strdup here? pool? or not keep regname at all?
-	var->type = strdup(type);
+	var->type = type;
 	var->kind = kind;
 	var->isarg = isarg;
 	var->delta = delta;
 	shadow_var_struct_members(var);
 	return var;
 }
 
-RZ_API void rz_analysis_var_set_type(RzAnalysisVar *var, const char *type) {
-	char *nt = strdup(type);
-	if (!nt) {
-		return;
-	}
-	free(var->type);
-	var->type = nt;
+RZ_API void rz_analysis_var_set_type(RzAnalysisVar *var, RzType *type) {
+	var->type = type;
```

Comment:
Add `RZ_OWN` to the `RzType *type` argument to indicate that ownership is transferred.